app/components/primer/popover_component.rb

```diff
@@ -5,37 +5,83 @@ module Primer
   #
   # By default, the popover renders with absolute positioning, meaning it should usually be wrapped in an element with a relative position in order to be positioned properly. To render the popover with relative positioning, use the relative property.
   class PopoverComponent < Primer::Component
-    include ViewComponent::Slotable
+    include ViewComponent::SlotableV2
 
-    with_slot :heading, class_name: "Heading"
-    with_slot :body, class_name: "Body"
+    CARET_DEFAULT = :top
+    CARET_MAPPINGS = {
+      CARET_DEFAULT => "",
+      :bottom => "Popover-message--bottom",
+      :bottom_right => "Popover-message--bottom-right",
+      :bottom_left => "Popover-message--bottom-left",
+      :left => "Popover-message--left",
+      :left_bottom => "Popover-message--left-bottom",
+      :left_top => "Popover-message--left-top",
+      :right => "Popover-message--right",
+      :right_bottom => "Popover-message--right-bottom",
+      :right_top => "Popover-message--right-top",
+      :top_left => "Popover-message--top-left",
+      :top_right => "Popover-message--top-right"
+    }.freeze
+
+    # The heading
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :heading, lambda { |**system_arguments|
+      system_arguments[:mb] ||= 2
+      system_arguments[:tag] ||= :h4
+
+      Primer::HeadingComponent.new(**system_arguments)
+    }
+
+    # The body
+    #
+    # @param caret [Symbol] <%= one_of(Primer::PopoverComponent::CARET_MAPPINGS.keys) %>
+    # @param large [Boolean] Whether to use the large version of the component.
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :body, lambda { |caret: CARET_DEFAULT, large: false, **system_arguments|
+      system_arguments[:classes] = class_names(
+        system_arguments[:classes],
+        "Popover-message Box",
+        CARET_MAPPINGS[fetch_or_fallback(CARET_MAPPINGS.keys, caret, CARET_DEFAULT)],
+        "Popover-message--large" => large
+      )
+      system_arguments[:p] ||= 4
+      system_arguments[:mt] ||= 2
+      system_arguments[:mx] ||= :auto
+      system_arguments[:text_align] ||= :left
+      system_arguments[:box_shadow] ||= :large
+
+      # This is a hack to allow the parent to set the slot's content
+      @body_arguments = system_arguments
+      ContentComponent.new
```

Comment:
```suggestion
    renders_one :body, lambda { |caret: CARET_DEFAULT, large: false, **system_arguments, &block|
      system_arguments[:classes] = class_names(
        system_arguments[:classes],
        "Popover-message Box",
        CARET_MAPPINGS[fetch_or_fallback(CARET_MAPPINGS.keys, caret, CARET_DEFAULT)],
        "Popover-message--large" => large
      )
      system_arguments[:p] ||= 4
      system_arguments[:mt] ||= 2
      system_arguments[:mx] ||= :auto
      system_arguments[:text_align] ||= :left
      system_arguments[:box_shadow] ||= :large

      # This is a hack to allow the parent to set the slot's content
      @body_arguments = system_arguments
      block.call
```

I *think* that would work, and be equivalent?

Reply:
will try